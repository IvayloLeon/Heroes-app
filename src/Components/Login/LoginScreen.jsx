import React, { useContext, useState } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const LoginScreen = ({history}) => {

    const [inputValue, setInputValue] = useState('');

    const {dispatch} = useContext(AuthContext);

    const handleLogin = () => {

        const lastPath = localStorage.getItem('lastPath') || '/';

        const data = {
            name: inputValue
        }

        const action = {
            type: types.login,
            payload: data
        }

        dispatch(action);

        history.replace(lastPath);
    }

    return (
        <div className='container mt-5'>
            <h1>Login</h1>
            <hr />

            <input 
                type="text" 
                className='form-control'
                placeholder='Ingrese aquÃ­ su nombre'
                value={inputValue}
                onChange={e => setInputValue(e.target.value)}    
            />

            <br />
            <br />

            <button
                className='btn btn-primary'
                onClick={handleLogin}
            >
                Login
            </button>
        </div>
    )
}
