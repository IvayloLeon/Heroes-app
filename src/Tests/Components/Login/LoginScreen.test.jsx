import React from 'react';
import { LoginScreen } from '../../../Components/Login/LoginScreen';
import {mount} from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../../auth/AuthContext';

import '@testing-library/jest-dom';

const history = {
    replace: jest.fn(),
}

const data = {
    dispatch: jest.fn()
}

test('Should to show (Snapshot) ', () => {
    const wrapper = mount(
        <MemoryRouter>
            <AuthContext.Provider value={data}>
                <LoginScreen history={history} />
            </AuthContext.Provider>
        </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
});

test('Should to do the dispatch and the navigation', () => {
    const wrapper = mount(
        <MemoryRouter>
            <AuthContext.Provider value={data}>
                <LoginScreen history={history} />
            </AuthContext.Provider>
        </MemoryRouter>
    );

    wrapper.find('.btn').simulate('click');
    expect(data.dispatch).toHaveBeenCalled();
    expect(history.replace).toHaveBeenCalled();
    expect(history.replace).toHaveBeenCalledWith('/');

    localStorage.setItem('lastPath', '/dc');
    wrapper.find('.btn').simulate('click');
    expect(history.replace).toHaveBeenCalledWith('/dc');

});