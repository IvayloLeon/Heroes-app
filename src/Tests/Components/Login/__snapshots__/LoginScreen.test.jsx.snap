// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should to show (Snapshot)  1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Ingrese aquÃ­ su nombre"
    type="text"
    value=""
  />
  <br />
  <br />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
